quill-sql/src/main/scala/io/getquill/context/sql/norm/ExpandDistinct.scala

```diff
@@ -12,6 +12,16 @@ object ExpandDistinct {
         Transform(q) {
           case Aggregation(op, Distinct(q)) =>
             Aggregation(op, Distinct(apply(q)))
+          case Distinct(Map(q, x, cc @ Tuple(values))) =>
+            Map(Distinct(Map(q, x, cc)), x,
+              Tuple(values.zipWithIndex.map {
+                case (_, i) => Property(x, s"_${i + 1}")
+              }))
+          case Distinct(Map(q, x, cc @ CaseClass(values))) =>
+            Map(Distinct(Map(q, x, cc)), x,
+              CaseClass(values.map {
+                case (str, _) => (str, Property(x, str))
+              }))
```

Comment:
I think the way this was before was a little naive. If a distinct block is returning a tuple like `Tuple(List(Property(Ident("foo"),"bar", ...))` and `Ident("returntable")` (especially if it's the same as `Ident("foo")` is being returned) you have to point back to that thing in the outer `Map` by doing something like `Tuple(List(Property(Ident("returntable"),"_1", ...))`. Same kind of thing for a case class. If inside `Distinct` it's `CaseClass(List("propname", Property(Ident("foo"),"bar"), ...)` you have to do something like `CaseClass(List("propname", Property(Ident("returntable"),"propname"), ...)`.

Reply:
LGTM

Reply:
It seems that this class doesn't have unit tests, maybe it's worth creating a few tests for it

Reply:
Yeah. Added `ExpandDistinctSpec`